electricitymap/contrib/lib/models/events.py

```diff
@@ -573,11 +573,20 @@ class Price(Event):
     currency: str
 
     @validator("currency")
-    def _validate_currency(cls, v: str):
+    def _validate_currency(cls, v: str) -> str:
         if v not in VALID_CURRENCIES:
             raise ValueError(f"Unknown currency: {v}")
         return v
 
+    @validator("datetime")
+    def _validate_datetime(cls, v: dt.datetime) -> datetime:
+        """Prices are given for the day ahead, so we should allow them to be in the future."""
+        if v.tzinfo is None:
+            raise ValueError(f"Missing timezone: {v}")
+        if v < LOWER_DATETIME_BOUND:
+            raise ValueError(f"Date is before 2000, this is not plausible: {v}")
+        return v
```

Comment:
This goes for the other events as well but should we add a check to make sure the datetime is actually a datetime here, before it's checked by the other validation?
```suggestion
    @validator("datetime")
    def _validate_datetime(cls, v: dt.datetime) -> datetime:
        """Prices are given for the day ahead, so we should allow them to be in the future."""
        if not isinstance(v, datetime):
            raise TypeError(f"Date is not a native datetime object: {v}")
        if v.tzinfo is None:
            raise ValueError(f"Missing timezone: {v}")
        if v < LOWER_DATETIME_BOUND:
            raise ValueError(f"Date is before 2000, this is not plausible: {v}")
        return v
```

Reply:
As far as I'm aware pydentic enforces the type before running the validation

Reply:
Just did a quick check and it seems to allow more than just datetimes:
`invalid type; expected datetime, string, bytes, int or float (type=type_error)`
But then again it also seems to convert non datetime types like timestamps to a datetime so it might not be a huge concern.

Reply:
Yes so actually it allows a lot of different things that are accepted but they all will get converted to datetimes.
https://docs.pydantic.dev/dev-v2/usage/types/datetime/
I would suggest for now not changing this, I will open another PR to make improvements there.

Reply:
Sounds good, I'll approve this so we can merge it.